Misc editorial improvements

ptomato opened the pull request:
~~Marked as draft, because I would like to write tests where appropriate before merging.~~

See: #1411

ljharb commented on spec/abstractops.html:
```diff
@@ -57,47 +57,19 @@ <h1>GetOption ( _options_, _property_, _type_, _values_, _fallback_ )</h1>
       1. Let _value_ be ? Get(_options_, _property_).
       1. If _value_ is *undefined*, return _fallback_.
-      1. Assert: _type_ is *"boolean"* or *"string"*.
+      1. Assert: _type_ is *"boolean"*, *"string"*, *"number"*, or *"stringOrNumber"*.
```
rather than `stringOrNumber`, why not accept a List of valid types? I think we do this in a couple places already.

ptomato replied:
Indeed, as in CreateListFromArrayLike? I thought about that, but here `stringOrNumber` is not just allowing either a string or a number, but has to be specific about calling ToString if the value is neither a number or a string.

Maybe having the List of valid types be in order of priority would work? I'll check if that makes things clearer / more general or not.

ljharb replied:
hm, it's a bit tricky if the valid type translates to a coercion algorithm - having `string` or `number` implicitly map to `ToString` or `ToNumber` is awkward, and i definitely see why you'd have gone with `stringOrNumber`.

ptomato replied:
I've done this in https://github.com/tc39/proposal-temporal/pull/1505/commits/ff399a30e119f40c2c978f0952641061bcd5bdc5 but kept it separate for now, I don't have a strong opinion on one being better than the other.

ljharb replied:
Nice, that seems like an improvement to me.

ljharb commented on spec/instant.html:
```diff
@@ -65,7 +65,7 @@ <h1>Temporal.Instant.fromEpochSeconds ( _epochSeconds_ )</h1>
         1. Set _epochSeconds_ to ? ToNumber(_epochSeconds_).
         1. Set _epochSeconds_ to ? NumberToBigInt(_epochSeconds_).
         1. Let _epochNanoseconds_ be _epochSeconds_ × 1,000,000,000.
```
i'm not sure this math works, unless 1e9 is written with the bigint notation?
```suggestion
        1. Let _epochNanoseconds_ be _epochSeconds_ × 1000000000<sub>ℤ</sub>.
```

ljharb commented on spec/instant.html:
```diff
@@ -80,7 +80,7 @@ <h1>Temporal.Instant.fromEpochMilliseconds ( _epochMilliseconds_ )</h1>
         1. Set _epochMilliseconds_ to ? ToNumber(_epochMilliseconds_).
         1. Set _epochMilliseconds_ to ? NumberToBigInt(_epochMilliseconds_).
         1. Let _epochNanoseconds_ be _epochMilliseconds_ × 1,000,000.
```
also here
```suggestion
        1. Let _epochNanoseconds_ be _epochMilliseconds_ × 1000000<sub>ℤ</sub>.
```

ljharb commented on spec/instant.html:
```diff
@@ -94,7 +94,7 @@ <h1>Temporal.Instant.fromEpochMicroseconds ( _epochMicroseconds_ )</h1>
       <emu-alg>
         1. Set _epochMicroseconds_ to ? ToBigInt(_epochMicroseconds_).
         1. Let _epochNanoseconds_ be _epochMicroseconds_ × 1,000.
```
and here
```suggestion
        1. Let _epochNanoseconds_ be _epochMicroseconds_ × 1000<sub>ℤ</sub>.
```

ljharb commented on spec/instant.html:
```diff
@@ -482,30 +480,24 @@ <h1>Properties of Temporal.Instant Instances</h1>
-        1. Assert: Type(_ns_) is BigInt.
-        1. If _ns_ &lt; −8.64 × 10<sup>21</sup><sub>ℝ</sub> or _ns_ &gt; 8.64 × 10<sup>21</sup><sub>ℝ</sub>, then
+        1. Assert: Type(_epochNanoseconds_) is BigInt.
+        1. If _epochNanoseconds_ &lt; −8.64 × 10<sup>21</sup><sub>ℝ</sub> or _epochNanoseconds_ &gt; 8.64 × 10<sup>21</sup><sub>ℝ</sub>, then
```
the R notation
```suggestion
        1. If _epochNanoseconds_ &lt; −8.64 × 10<sup>21</sup><sub>ℤ</sub> or _epochNanoseconds_ &gt; 8.64 × 10<sup>21</sup><sub>ℤ</sub>, then
```

ljharb commented on spec/instant.html:
```diff
@@ -567,7 +559,7 @@ <h1>AddInstant ( _epochNanoseconds_, _hours_, _minutes_, _seconds_, _millisecond
             _seconds_ × 1,000,000,000 +
             _minutes_ × 60,000,000,000 +
             _hours_ × 3,600,000,000,000.
```
`AddInstant` needs to add type assertions to its arguments, and also be explicit about which kind of numbers are being used in this equation.

ljharb commented on spec/instant.html:
```diff
@@ -64,8 +64,8 @@ <h1>Temporal.Instant.fromEpochSeconds ( _epochSeconds_ )</h1>
         1. Set _epochSeconds_ to ? NumberToBigInt(_epochSeconds_).
-        1. Let _epochNanoseconds_ be _epochSeconds_ × 1,000,000,000.
-        1. Perform ? RejectInstant(_epochNanoseconds_).
+        1. Let _epochNanoseconds_ be _epochSeconds_ × *1,000,000,000*<sub>ℤ</sub>.
```
the spec doesn't currently use any `,` separators in long numbers (like 1000).
```suggestion
        1. Let _epochNanoseconds_ be _epochSeconds_ × *1000000000*<sub>ℤ</sub>.
```

ptomato replied:
OK, in that case I've gone with scientific notation — I'd assume we don't want people to make mistakes counting the zeroes in 3600000000000.

ljharb commented on spec/instant.html:
```diff
@@ -561,13 +553,14 @@ <h1>AddInstant ( _epochNanoseconds_, _hours_, _minutes_, _seconds_, _millisecond
+            ℤ(_microseconds_) × *1000*<sub>ℤ</sub> +
+            ℤ(_milliseconds_) × *10<sup>6</sup>*<sub>ℤ</sub> +
+            ℤ(_seconds_) × *10<sup>9</sup>*<sub>ℤ</sub> +
+            ℤ(_minutes_) × *6 × 10<sup>10</sup>*<sub>ℤ</sub> +
+            ℤ(_hours_) × *3.6 × 10<sup>12</sup>*<sub>ℤ</sub>.
```
```suggestion
            ℤ(_minutes_) × *6<sub>ℤ</sub> × 10<sup>10</sup>*<sub>ℤ</sub> +
            ℤ(_hours_) × *3.6<sub>ℤ</sub> × 10<sup>12</sup>*<sub>ℤ</sub>.
```

ljharb replied:
ah right, duh. Either of your suggestions is fine; as long as every number is in the same space.

ptomato replied:
I'll go with 3600<sub>ℤ</sub> × 10<sup>9</sup><sub>ℤ</sub> since if we are not doing proper scientific notation, this is the most illustrative way to write the multiplication in my opinion.

ljharb reviewed (approved):
Spec seems good to me!

Ms2ger commented on spec/abstractops.html:
```diff
@@ -45,59 +45,35 @@ <h1>GetOptionsObject ( _options_ )</h1>
-        1. Let _value_ be ! ToBoolean(_value_).
-      1. If _type_ is *"string"*, then
-        1. Let _value_ be ? ToString(_value_).
+      1. If _types_ contains an element equal to Type(_value_), then
```
```suggestion
      1. If _types_ contains Type(_value_), then
```

ptomato commented:
What's changed since the last time: I removed the GetOption commit since I'm still working on ensuring that all the entry points that use GetOption are covered by test262 tests. I added test262 tests that cover "Remove NonNegativeModulo abstract operation". This is ready for merging if the test262 tests pass review. I'll open a separate PR for the GetOption changes so that this one isn't delayed on that any longer.
